### PR TITLE
Fixed the implementation of getMap.

### PR DESCRIPTION
**Pull request by TikhonJelvis**
The previous version of getMap did not handle blocks that began with a negative number. (See #85 for details.) This PR fixes the behavior and shares the code to handle blocks between `getMap` and `getArray`.

I'm pretty sure we need a similar fix in `Decode.Lazy` and `Decode.Strict`, but I don't fully understand what's going on in those modules and don't really want to touch them right now.

### CODE REVIEW

**Comment by AlexeyRaga** [comment]
Thanks @TikhonJelvis 
Do you think that https://github.com/haskell-works/avro/blob/master/src/Data/Avro/Decode/Lazy.hs#L269 suffers from it too (or elsewhere in the lazy decoder)?

**Comment by TikhonJelvis** [comment]
I'm pretty sure it does, but I'm finding it hard to follow that code right now, so I'd need more time to understand what's going on and figure out how to fix it.

**Comment by TikhonJelvis** [comment]
Wait, never mind, the "blocks" used to encode _containers_ are different than the blocks used to encode maps and arrays. The `decodeBlock` function you linked is working with the blocks of a container, so it cannot have this same problem.

The same problem might exist in the [`getAvroOf`](https://github.com/haskell-works/avro/blob/master/src/Data/Avro/Decode/Lazy.hs#L286) implementation, but I'd need to spend more time with the code to know for sure.

**Comment by AlexeyRaga** [comment]
@TikhonJelvis can we merge this or you are still looking at the edge cases?

**Comment by TikhonJelvis** [comment]
Oh, sorry, forgot about this PR. The PR itself is definitely mergeable—I was just worried that some of the code I _didn't_ touch had the same kind of bug. I'm still not sure whether that's the case, but we can merge this without worrying about that.
